src/pages/index.js

```diff
@@ -103,70 +116,176 @@ const settings = {
 const editFormElement = editProfileModal.querySelector(".modal__form");
 const addFormElement = addCardModal.querySelector(".modal__form");
 
+const avatarModalPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (avatarData) => {
+    avatarModalPopup.setButtonText(true);
+    api
+      .editUserPhoto(avatarData)
+      .then((data) => {
+        userInfo.setAvatar(data.avatar);
+        avatarModalPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => avatarModalPopup.setButtonText());
+  }
+);
+
+const avatarModalElement = avatarModal.querySelector(".modal__form");
+
+/*----------------------------------*/
+/*      card like/remove likes      */
+/*----------------------------------*/
+
+// const handleLikeCard = (card) => {
+//   if (card._isliked) {
+//     api
+//       .removeCardLike(card._cardId)
+//       .then(() => {
+//         card.setLikes(false);
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   } else {
+//     api
+//       .addCardLike(card._cardId)
+//       .then(() => {
+//         card.setLikes(true);
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   }
+// };
+
 /*----------------------*/
 /*      popup items     */
 /*----------------------*/
 
-const editFormValidator = new FormValidator(settings, editFormElement);
-const addFormValidator = new FormValidator(settings, addFormElement);
-
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
-
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
+const addCardPopup = new PopupWithForm("#add-card-modal", (cardData) => {
+  addCardPopup.setButtonText(true);
+  return api
+    .addCard(cardData)
+    .then((res) => {
+      console.log(res);
+      createCard(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 addCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileFormSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (card) => {
+  editProfilePopup.setButtonText(true);
+  return api
+    .editUserInfo(card)
+    .then((res) => {
+      console.log(res);
+      userInfo.setUserInfo(res.name, res.about);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 editProfilePopup.setEventListeners();
 
-//function renderCard(cardData, wrapper) {
-//const newCard = new Card(cardData, cardSelector, handleCardClick);
-//cardSection.addItem(newCard.getView());
-//}
+avatarModalPopup.setEventListeners();
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    renderer: createCard,
-  },
-  cardListSelector
-);
+const editFormValidator = new FormValidator(settings, editFormElement);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object `formValidators`.  And then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
const formValidators = {}

// enable validation
const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // here you get the name of the form
    const formName = formElement.getAttribute('name')

   // here you store a validator by the `name` of the form
    formValidators[formName] = validator;
   validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```